frontend/catalyst/device/decomposition.py

```diff
@@ -47,18 +47,112 @@
 logger.addHandler(logging.NullHandler())
 
 
+def check_alternative_control_support(op, capabilities):
+    """Verify that aliased controlled operations aren't supported via alternative definitions."""
+
+    if (
+        isinstance(op, qml.ControlledQubitUnitary)
+        and capabilities.native_ops.get("QubitUnitary")
+        and capabilities.native_ops.get("QubitUnitary").controllable
+    ):
```

Comment:
There is probably a better way to this than long if chains but oh well 😅

Reply:
Omg is it that simple? 😅

Reply:
Yeah earlier this year I worked to make all custom controlled ops inherit from `Controlled`, which should work.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1002